lib/Doctrine/DBAL/Schema/SchemaDiff.php

```diff
@@ -139,9 +139,15 @@ protected function _toSql(AbstractPlatform $platform, $saveMode = false)
 
         $foreignKeySql = [];
         foreach ($this->newTables as $table) {
+            $tableCreateSQL = $platform->getCreateTableSQL($table, AbstractPlatform::CREATE_INDEXES);
+
+            if (empty($tableCreateSQL)) {
```

Comment:
IMO the whole event-listener model used to extend the platform behavior is quite clumsy. Events handlers should be able to augment or prevent certain actions but they shouldn't _alter_ the underlying system's behavior.

The very fact that the`_getCreateTableSQL()` can return an empty array violates its own contract: https://github.com/doctrine/dbal/blob/5dd30ad24dde117ca0f272d370dbafe30fc2a5a1/src/Platforms/AbstractPlatform.php#L1714-L1715

If this way of extending the platform functionality is considered valid, then all consumers of this API should account for empty SQL potentially returned from this method which is an unnecessary complication.